vra-services-check/src/testsuite_module/vrahealthtestsuite.py

```diff
@@ -0,0 +1,236 @@
+            
+        else:
+            #if NULL - return false
+            #print("No registered vRA services were retrieved - FAILED HEALTH TEST!")
+            assert False, "vRA Registered Services were not retrieved - FAILING HEALTH TEST!"
+
+        return True
+    #end of Test suite method 'check_registered_services' definition
+	
+    #Start function definition to obtain distinct endpoint types in vRA tenant
+    def get_distinct_endpoint_types(self,url,user,pwd,tenant):
+         
+         #first, login to VRA specific tenant identified by 'tenant'
+         theToken = self.login_vra_token(url + "/identity/api/tokens", user, pwd, tenant)
+         
+         if theToken != None:
+             headers = {'Accept': "application/json", 'Authorization': "Bearer " + theToken,'Cache-Control': "no-cache"}
+             querystring = {"page": "1", "limit": "20"}
+             response = requests.request("GET", url+"/component-registry/endpoints/types", headers=headers, params=querystring, verify=False)
+             if response:
+                 #print("DEBUG SO FAR OK - Obtained vRA Distinct Endpoint types response \n")
+                 # servicesJSON = json.loads(response.content)
+                 endpointsJSON = json.loads(response.content.decode('utf-8'))
+                 #print("DEBUG OK - parsed Distinct Endpoint types into JSON format! \n")
+                 return endpointsJSON['content']
+             else:
+                 #print("DEBUG, PROBLEM: failed to obtain Distinct Endpoint types!")
+                 #return response
+                 raise Exception("Failed to obtain distinct vRA Endpoint types for tenant: "+tenant)
+
+         else:
+             #print("DEBUG, PROBLEM: vRA login failed - cannot retrieve distinct Endpoints data!")
+             return None
+
+    #end function definition to obtain distinct endpoint types in vRA tenant
+
+    #start Test suite method 'check_distinct_endpoints' definition
+    @test_name("Check number of vRA Distinct Endpoints")
+    @test_description("Checks that expected number of Distinct vRA Endpoints is available")
+    @test_access_level(AccessLevel.NORMAL.value)
+    @test_severity(Severity.NORMAL.value)
+    @test_remediation("https://www.vmware.com/products/vrealize-automation.html")
+    @test
+    def check_distinct_endpoints(self, vraaddress, vrausername, vrapassword, vratenant, numendpoints):
+        
+        endpointsJSON = self.get_distinct_endpoint_types(vraaddress, vrausername, vrapassword, vratenant)
+
+        if endpointsJSON:
+            #print("Asserting length of all Distinct VRA endpoints >= " + str(numendpoints) + " - " + str(len(endpointsJSON) >= int(numendpoints)))
+            assert len(endpointsJSON) >= int(numendpoints), "Number of Distinct vRA endpoints: " + str(len(endpointsJSON)) + " is less than expected: " + str(numendpoints)
```

Comment:
Is there a way to display which endpoint is actually missing?

Reply:
Not at the moment as I'm just counting the endpoints and comparing against the target count I can work on adding that later

Reply:
ok